File: src/main/java/ar/edu/unq/unidad1/wop/modelo/exception/MuchoPesoException.java
package ar.edu.unq.unidad1.wop.modelo.exception;

import ar.edu.unq.unidad1.wop.modelo.Item;
import ar.edu.unq.unidad1.wop.modelo.Personaje;

public class MuchoPesoException extends RuntimeException {
    public MuchoPesoException(Personaje personaje, Item item) {
        super("El personaje [" + personaje + "] no puede recoger [" + item + "] porque cagar mucho peso ya");
    }
}

File: src/main/java/ar/edu/unq/unidad1/wop/modelo/exception/MuchoPesoException.kt
package ar.edu.unq.unidad1.wop.modelo.exception

import ar.edu.unq.unidad1.wop.modelo.Item
import ar.edu.unq.unidad1.wop.modelo.Personaje

class MuchoPesoException(private val personaje: Personaje, private val item: Item) :
    RuntimeException() {
    override val message: String
        get() = "El personaje [" + personaje + "] no puede recoger [" + item + "] porque cagar mucho peso ya"

    companion object {
        private const val serialVersionUID = 1L
    }

}